src/snowflake/snowpark/column.py

```diff
@@ -225,108 +238,359 @@ class Column:
     This class has methods for the most frequently used column transformations and operators. Module :mod:`snowflake.snowpark.functions` defines many functions to transform columns.
     """
 
+    # NOTE: For now assume Expression instances can be safely ignored when building AST
+    #       Expression logic can be eliminated entirely once phase 0 is integrated
+    #       Currently a breaking example can be created using the Column.isin method as it does not build the AST.
+    #       For example, running: df.filter(col("A").isin(1, 2, 3) & col("B")) would fail since the boolean operator
+    #       '&' would try to construct an AST using that of the new col("A").isin(1, 2, 3) column (which we currently
+    #       don't fill if the only argument provided in the Column constructor is 'expr1' of type Expression)
     def __init__(
-        self, expr1: Union[str, Expression], expr2: Optional[str] = None
+        self,
+        expr1: Union[str, Expression],
+        expr2: Optional[str] = None,
+        ast: Optional[proto.Expr] = None,
```

Comment:
Is `ast` here for internal use only?

Reply:
+1. can we make this private

Reply:
addressed in https://github.com/snowflakedb/snowpark-python/pull/2507/files